lib/igpumonitor.h

```diff
@@ -26,17 +28,10 @@ class IGPUMonitor
 public:
     virtual ~IGPUMonitor() {}
 
-    // Start monitoring
-    virtual void start(int period) = 0;
-    // Stop monitoring
-    virtual void stop() = 0;
-    // Return if monitor is currently watching data
-    virtual bool watching() const = 0;
-
     // Usage should be in percentage
-    virtual float getUsage() const = 0;
+    virtual const std::vector<float>& getUsage() = 0;
```

Comment:
Ok with the API changes. can you update the README accordingly with the new signatures:

"""

To monitor a specific GPU, you must subclass `IGPUMonitor`:
```cpp
#include <uprofile/igpumonitor.h>

class MyGPUMonitor: public uprofile::IGPUMonitor {
public:
    float getUsage() override;
    void getMemory(int& usedMem, int& totalMem) override;
}
```
"""